controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,333 @@
+
+	odfDepsCsv := &oprv1a1.ClusterServiceVersion{}
+
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: OdfDepsSubscriptionStartingCSV, Namespace: r.OperatorNamespace}, odfDepsCsv)
+	if err != nil {
+		r.log.Error(err, "failed getting odf-deps csv", "csvName", OdfDepsSubscriptionStartingCSV)
+		return err
+	}
+
+	if odfDepsCsv.Status.Phase != oprv1a1.CSVPhaseSucceeded {
+		err = fmt.Errorf("csv %s is not in succeeded state", OdfDepsSubscriptionStartingCSV)
+		r.log.Error(err, "waiting for csv to be in succeeded state")
+		return err
+	}
+
+	return nil
+}
+
+func (r *OperatorScalerReconciler) reconcileOperators() error {
+
+	var returnErr error
+
+	for _, metaData := range MetaDataList {
+
+		crList := &metav1.PartialObjectMetadataList{}
+		crList.TypeMeta.APIVersion = metaData.ApiVersion
+		crList.TypeMeta.Kind = metaData.Kind
+
+		err := r.Client.List(r.ctx, crList)
+		if err != nil {
+			if !meta.IsNoMatchError(err) {
+				msg := fmt.Sprintf("failed listing %s", metaData.Kind)
+				r.log.Error(err, msg)
+				multierr.AppendInto(&returnErr, err)
+			}
+		} else {
+			if len(crList.Items) >= 0 {
+
+				for _, pkgName := range metaData.PkgNames {
+					key := fmt.Sprintf(CsvLabelKey, pkgName, r.OperatorNamespace)
+
+					csvList := &oprv1a1.ClusterServiceVersionList{}
+					err = r.Client.List(
+						r.ctx, csvList,
+						client.InNamespace(r.OperatorNamespace),
+						client.MatchingLabels(map[string]string{key: ""}),
+					)
+					if err != nil {
+						r.log.Error(err, "failed listing csvs with label", "label", key)
+						multierr.AppendInto(&returnErr, err)
+					} else {
+						err = r.updateCsvsReplicas(csvList.Items...)
+						if err != nil {
+							r.log.Error(err, "failed updating csvs replica")
+							multierr.AppendInto(&returnErr, err)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) updateCsvsReplicas(csvList ...oprv1a1.ClusterServiceVersion) error {
+
+	var returnErr error
+
+	for idx := range csvList {
+		csv := &csvList[idx]
+		var updateRequired bool
+		for i := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+			if *csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec.Replicas != int32(1) {
+				updateRequired = true
+			}
+			csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec.Replicas = ptr.To(int32(1))
+		}
+
+		if updateRequired {
+			err := r.Client.Update(r.ctx, csv)
+			if err != nil {
+				r.log.Error(err, "failed updating csv replica", "csvName", csv.Name)
+				multierr.AppendInto(&returnErr, err)
+			} else {
+				r.log.Info("csv updated successfully", "csvName", csv.Name)
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) reconcileDynamicWatchers(crdName string) error {
+
+	for _, metaData := range MetaDataList {
+		if !r.watching[metaData.CrdName] {
+
+			crd := &extv1.CustomResourceDefinition{}
+			err := r.Client.Get(r.ctx, types.NamespacedName{Name: metaData.CrdName}, crd)
+			if err != nil {
```

Comment:
Prefer locality of error variables
```suggestion
			if err := r.Client.Get(r.ctx, types.NamespacedName{Name: metaData.CrdName}, crd); err != nil {
```

Reply:
Done